platform/state/platform_state.proto

```diff
@@ -0,0 +1,195 @@
+
+    // Fields below are to be deprecated in the foreseeable future.
+
+    /**
+     * A running event hash.<br/>
+     * This is computed by the consensus event stream.
+     * <p>
+     * This will be _removed_ and the field number reserved once the consensus
+     * event stream is retired.
+     */
+     bytes legacy_running_event_hash = 10000 [deprecated = true];
+
+    /**
+     * A consensus generation.<br/>
+     * The lowest judge generation before birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 lowest_judge_generation_before_birth_round_mode = 10001 [deprecated = true];
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * Will be removed after the birth round migration.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 last_round_before_birth_round_mode = 10002 [deprecated = true];
+
+    /**
+     * A consensus node semantic version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * If birth round migration is complete, this SHALL be the _first_ software
+     * version that enabled birth round mode.
+    */
+    proto.SemanticVersion first_version_in_birth_round_mode = 10003 [deprecated = true];
+
+}
+
+
+/**
+ * A consensus snapshot.<br/>
+ * This is a snapshot of the consensus state for a particular round.
+ *
+ * This message SHALL record consensus data necessary for restart
+ * and reconnect.
+ */
+message ConsensusSnapshot {
+    /**
+     * A consensus round.<br/>
+     * The round number of this snapshot.
+     */
+    uint64 round = 1;
+    /**
+     * A list of SHA-384 hash values.<br/>
+     * The hashes of all judges for this round.
+     * <p>
+     * This list SHALL be ordered by creator ID.<br/>
+     * This list MUST be deterministically ordered.
+     */
+    repeated bytes judge_hashes = 2;
+
+    /**
+     * A list of minimum judge information entries.<br/>
+     * These are "minimum ancient" entries for non-ancient rounds.
+     */
+    repeated MinimumJudgeInfo minimum_judge_info_list = 3;
+
+    /**
+     * A single consensus number.<br/>
+     * The consensus order of the next event to reach consensus.
+     */
+    uint64 next_consensus_number = 4;
+
+    /**
+     * A "consensus" timestamp.<br/>
+     * The consensus timestamp of this snapshot.
+     * <p>
+     * Depending on the context this timestamp may have different meanings:
+     * <li> if there are transactions, the timestamp is equal to the timestamp of the last transaction
+     * <li> if there are no transactions, the timestamp is equal to the timestamp of the last event
+     * <li> if there are no events, the timestamp is equal to the timestamp of the previous round plus a small constant
```

Comment:
So these are incorrect HTML and won't process correctly (even Javadoc's broken HTML implementation might not like an improper list setup like this).
It can be fixed separately (this PR has already taken a long time), however.
```suggestion
     * Depending on the context this timestamp may have different meanings:
     * <ul>
     * <li>if there are transactions, the timestamp is equal to the timestamp of the last transaction</li>
     * <li>if there are no transactions, the timestamp is equal to the timestamp of the last event</li>
     * <li>if there are no events, the timestamp is equal to the timestamp of the previous round plus a small constant</li>
     * </ul>
```